VDF.Core/ScanEngine.cs

```diff
@@ -214,35 +214,53 @@ public sealed class ScanEngine {
 					while (pauseTokenSource.IsPaused) Thread.Sleep(50);
 
 					var entry = ScanList[i];
+					float[] percent = new float[positionList.Count];
+					Dictionary<double, byte[]?>[] grayBytes = new Dictionary<double, byte[]?>[2];
```

Comment:
The base idea of crating the array was to combine the two cases (flip or not) retrospectively by a simple loop so that the actual functionality remains clear and the two cases do not have to be distinguished again.

And yes, it would be possible to replace the `2` by `(Settings.CompareHorizontallyFlipped ? 2 : 1)`
But this has no effect on the image case. In the image case, in principle you would not need a dictionary. But then image an video case would need to be completely separated.

Reply:
I have now taken a different approach.
Instead of the loop and the additional array, the comparison is now outsourced to a function. 
This is the clearest way from my point of view.